### DISCUSSION TITLE
Frequency of CVE updates

### DISCUSSION THREAD

**Discussion by topstair** [quay/clair#1811] [Q&A] [answered]
How often does Clair poll sources for new vulnerabilities?

I've reviewed [the "Updater" config settings](https://github.com/quay/clair/blob/main/Documentation/reference/config.md#updaters), but was unable to find a value for update frequency.
I also checked the ["Updaters and Airgap" documentation](https://quay.github.io/clair/concepts/updatersandairgap.html) to no avail.

Similar products allow the update frequency to be set.
e.g. [Anchore has a default "feed_sync"](https://docs.anchore.com/current/docs/using/cli_usage/feeds/feed_configuration/) of 14400 seconds (e.g. update frequency of every 6 hr).
Is there a similar config setting in Clair?
Is the frequency hardcoded? If so, what is the frequency?
Or does Clair use a different paradigm? (e.g. maybe dynamically updates before every scan, so no need to specify frequency?)
Is there a way to confirm in the logs that the "updater" has been synced with a source?

Tags: timing, often, frequency, periodic, daily, hourly, polling, request, update

**Comment by crozzy** [comment, marked as answer]
Update frequency is determined by https://github.com/quay/clair/blob/main/Documentation/reference/config.md#matcherperiod. The default is every 6 hours (docs appear to be reporting the old default, will update).

**Reply by crozzy** [reply]
https://github.com/quay/clair/pull/1812
